### PR TITLE
[FSDP] Add support for `FSDPCommContext` on PrivateUse1 backend

### PR DESCRIPTION
**Pull request by shink**
I believe NPU supports streams, but what about other privateuse1 backends?

cc: @FFFrog @noemotiovon

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Comment by awgu** [comment]
I am ok with this change. Let me know if any of the PrivateUse1 do not support streams and have a comment about this.

File: torch/distributed/_composable/fsdp/_fsdp_param_group.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ class FSDPCommContext:
 
     def lazy_init(self, device: torch.device):
         self.device_handle = _get_device_handle(device.type)
-        if device.type not in ["cuda", "hpu"]:
+        if device.type not in ["cuda", "hpu", torch._C._get_privateuse1_backend_name()]:
```
Comment (shink) [inline torch/distributed/_composable/fsdp/_fsdp_param_group.py:54]: Do we have an API to detect if the given device supports streams? @albanD @FFFrog

Comment (kwen2501) [reply torch/distributed/_composable/fsdp/_fsdp_param_group.py:54]: @albanD @FFFrog thoughts?

Comment (FFFrog) [reply torch/distributed/_composable/fsdp/_fsdp_param_group.py:54]: Sorry for the late comments.
As far as I know, there is no some one API detecting if supports stream.

This PR can be closed now due to https://github.com/pytorch/pytorch/pull/139539

Comment (shink) [reply torch/distributed/_composable/fsdp/_fsdp_param_group.py:54]: Thanks for the info, I’m gonna close this PR.

**Pull request closed without merging**